### DISCUSSION TITLE
Map Context Editor: Layer with multiple styles

### DISCUSSION THREAD

**Discussion by urothstein** [mrmap-community/mrmap#349] [General]
If a layer offers multiple styles: which one will be associated with the added layer:

- default - so no style to request explicitly?
- must there be a selection?

**Comment by jokiefer** [comment]
As discussed today in the daily:
* style
* bbox
* width
* height
* ...
are client specific params. So the backend can only deliver the possible information per layer what the client could choice. All listed attributes are part of the jsonapi responses. 

I see the following soluitons:
* If a user adds a layer to the context, then the default style shall be used. This can be done with the empty query parameter for styles. No information from the backend is needed for that case.
* Optional a user can select an other style for that specific layer. Needed information can be picked up from the jsonapi response.

**Reply by armin11** [reply]
In sense of OWS Context the selected style is part of the given getmap url. It is also possible to  use the extensions of the StyleSet object to mark the selected style as active. Corresponding issue in old system: https://github.com/mrmap-community/Mapbender2.8/issues/91

**Reply by urothstein** [reply]
means that we want to offer a nested list when adding layers, i.e. all layers on the first level and then all styles on the second. Then when a style is selected, the layer is added to the tree and the corresponding style is marked as active for the context document?

**Reply by urothstein** [reply]
Example: https://ant.design/components/tree-select/

**Reply by urothstein** [reply]
One more point: do you have an example where multiple layers linked to one metadata set and also multiple styles are offered per layer?

**Reply by jokiefer** [reply]
services from dwd... but currently the metadataurl's are broken...

**Reply by urothstein** [reply]
https://ant.design/components/cascader/

**Comment by jokiefer** [comment]
![image](https://user-images.githubusercontent.com/17874616/148556421-40a7bddd-0e1f-4fcb-9024-3a41c90de018.png)
